cmd/clusterctl/client/alpha/rollout_rollbacker.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/util"
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+	"sigs.k8s.io/cluster-api/controllers/mdutil"
+	"sigs.k8s.io/cluster-api/util/patch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ObjectRollbacker will issue a rollback on the specified cluster-api resource.
+func (r *rollout) ObjectRollbacker(proxy cluster.Proxy, tuple util.ResourceTuple, namespace string, toRevision int64) error {
+	switch tuple.Resource {
+	case machineDeployment:
+		deployment, err := getMachineDeployment(proxy, tuple.Name, namespace)
+		if err != nil || deployment == nil {
+			return errors.Wrapf(err, "failed to fetch %v/%v", tuple.Resource, tuple.Name)
+		}
+		if deployment.Spec.Paused {
+			return errors.Errorf("can't rollback a paused MachineDeployment (run rollout resume first): %v/%v\n", tuple.Resource, tuple.Name)
+		}
+		if err := rollbackMachineDeployment(proxy, deployment, tuple.Name, namespace, toRevision); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("invalid resource type %q, valid values are %v", tuple.Resource, validResourceTypes)
+	}
+	return nil
+}
+
+// rollbackMachineDeployment will rollback to a previous MachineSet revision used by this MachineDeployment.
+func rollbackMachineDeployment(proxy cluster.Proxy, d *clusterv1.MachineDeployment, name, namespace string, toRevision int64) error {
```

Comment:
```suggestion
func rollbackMachineDeployment(proxy cluster.Proxy, d *clusterv1.MachineDeployment, toRevision int64) error {
```
given that we are already passing the MachineDeployment

Reply:
done